### PR TITLE
V5 virtual body fc

### PR DESCRIPTION
**Pull request by colinlohner**
## Overview 
 Update virtual body to FC/Hooks

## Screenshots (if applicable) 
N/a

## Testing
- [x] Existing functionality remains

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -1829,8 +1829,7 @@ export class Table extends React.PureComponent<TableProps> {
   public static Body = TableBody
 
   // @ts-ignore
```
Comment (mshwery) [inline index.d.ts:1831]: do we need the `@ts-ignore` still?

File: src/table/src/TableVirtualBody.js [hunk 2]
```diff
@@ -1,136 +1,84 @@
...
-    scrollOffset: PropTypes.number,
-    /**
-     * The scrollToAlignment property passed to react-tiny-virtual-list
-     */
-    scrollToAlignment: PropTypes.oneOf(['start', 'center', 'end', 'auto']),
-    /**
-     * The onScroll callback passed to react-tiny-virtual-list
-     */
-    onScroll: PropTypes.func
+const useForceUpdate = () => useState()[1]
```
Comment (mshwery) [inline src/table/src/TableVirtualBody.js:7]: I dont understand this

Comment (mshwery) [reply src/table/src/TableVirtualBody.js:7]: Are you trying something like this? https://stackoverflow.com/questions/53215285/how-can-i-force-component-to-re-render-with-hooks-in-react

Comment (colinlohner) [reply src/table/src/TableVirtualBody.js:7]: essentially yes, just trying to force a rerender

File: index.d.ts (hunk 1 above)
Comment (colinlohner) [reply index.d.ts:1831]: not sure, i'll make sure we do/dont

File: src/table/src/TableVirtualBody.js (hunk 2 above)
Comment (mshwery) [reply src/table/src/TableVirtualBody.js:7]: I'm not sure what you have will actually work. What you want is something like this:

```js
const useForceUpdate = () => {
  const [_, setValue] = useState()

  return useCallback(() => {
    setValue({}) // set with a new object instance every time it gets called
  }, [setValue])
}
```

Comment (colinlohner) [reply src/table/src/TableVirtualBody.js:7]: It does seem to work, but if you think that what you have there would work out better, I can update it to that

File: src/table/src/TableVirtualBody.js [hunk 3]
```diff
@@ -149,52 +97,44 @@ export default class TableVirtualBody extends PureComponent {
...
 
     // Simply return when we now the height of the pane is fixed.
-    if (this.state.isIntegerHeight) return
+    if (isIntegerHeight) return
 
     // Return if we are in a weird edge case in which the ref is no longer valid.
-    if (this.paneRef) {
-      const calculatedHeight = this.paneRef.offsetHeight
+    if (paneRef && paneRef instanceof Node) {
+      const calculatedHeight = paneRef.offsetHeight
```
[the remark this answers is not in the document]
Comment (colinlohner) [reply src/table/src/TableVirtualBody.js:133]: it is intended, but maybe i should change the variable name so there isnt confusion

File: src/table/src/TableVirtualBody.js (hunk 2 above)
Comment (mshwery) [reply src/table/src/TableVirtualBody.js:7]: I think you need to call set state with a new object for it to rerender

**Pull request merged**